File: SmartShop/app/src/main/java/com/example/smartshop/model/ChiTietDonHang.java
package com.example.smartshop.model;

public class ChiTietDonHang {
    private int id;
    private int maDonHang;
    private String maSanPham;
    private String tenSanPham;
    private int giaSanPham;
    private int soLuongSanPham;

    public ChiTietDonHang(int id, int maDonHang, String maSanPham, String tenSanPham, int giaSanPham, int soLuongSanPham) {
        this.id = id;
        this.maDonHang = maDonHang;
        this.maSanPham = maSanPham;
        this.tenSanPham = tenSanPham;
        this.giaSanPham = giaSanPham;
        this.soLuongSanPham = soLuongSanPham;
    }

    public int getId() {
        return id;
    }

    public int getMaDonHang() {
        return maDonHang;
    }

    public String getMaSanPham() {
        return maSanPham;
    }

    public String getTenSanPham() {
        return tenSanPham;
    }

    public int getGiaSanPham() {
        return giaSanPham;
    }

    public int getSoLuongSanPham() {
        return soLuongSanPham;
    }
}

File: ServerSmartShop/Main/chitietdonhang.php
<?php
    include "connect.php";
    $json = $_POST['json'];
    $data = json_decode($json, true);
    $all_success = true;

    foreach($data as $key => $value){
        $madonhang = $value['madonhang'];
        $masanpham = $value['masanpham'];
        $tensanpham = $value['tensanpham'];
        $giasanpham = $value['giasanpham'];
        $soluongsanpham = $value['soluongsanpham'];

        $query = "INSERT INTO chitietdonhang (id, madonhang, masanpham, tensanpham, giasanpham, soluongsanpham) 
                  VALUES (null, '$madonhang', '$masanpham', '$tensanpham', '$giasanpham', '$soluongsanpham')";
        $Dta = mysqli_query($conn, $query);

        if(!$Dta){
            $all_success = false;
        }
    }

    if($all_success){
        echo "1";
    }else{
        echo "0";
    }
?>
